packages/core/package.json

```diff
@@ -63,6 +63,8 @@
     "reflect-metadata": "^0.1.10",
     "route-parser": "^0.0.5",
     "safer-buffer": "^2.1.2",
+    "socket.io": "4.1.0",
```

Comment:
@msujew can we upgrade `socket.io` to the latest, this version currently pulls a vulnerable version of `engine.io`:
- [dependabot alert](https://github.com/eclipse-theia/theia/security/dependabot/18)

`engine.io` range in `socket.io`:
- [`socket.io@4.1.0`](https://github.com/socketio/socket.io/blob/fb6b0efec9fcc154dc9131189d78fcfa0a6c7f46/package.json#L54)
- [`socker.io@4.4.0`](https://github.com/socketio/socket.io/blob/c82a4bdf1f900644cd7ee823a32c7820164fe80e/package.json#L52)